test/lit.cfg

```diff
@@ -733,6 +733,65 @@ elif run_os == 'linux-gnu' or run_os == 'linux-gnueabihf' or run_os == 'freebsd'
     config.target_ld = (
         "ld -L%s" %
         (os.path.join(test_resource_dir, config.target_sdk_name)))
+elif run_os == 'linux-androideabi':
+    # Android
+    lit_config.note("Testing Android " + config.variant_triple)
+    # FIXME: This value should be read off of config.android_ndk_path,
+    #        but I can't figure out how. The value is set via CMake, and the
+    #        build directory contains a test-android-armv7/lit.site.cfg with
+    #        its @SWIFT_ANDROID_NDK_PATH@ correctly replaced. So what's wrong?
```

Comment:
When I invokve `utils/built-script -T`, the `lit.site.cfg.in` is correctly configured and placed in my build directory:

```py
# build/Ninja-ReleaseAssert/swift-linux-x86_64/test-android-armv7/lit.site.cfg
config.android_ndk_path = "/home/modocache/android-ndk-r10e"
```

But when the tests are run later during that same `utils/build-script -T` invocation, the attempt to access the value from within the `test/lit.cfg` fails:

```py
# swift/test/lit.cfg
android_ndk_path = config.android_ndk_path

# lit.py: lit.cfg:738: note: Testing Android armv7-none-linux-androideabi
# lit.py: TestingConfig.py:114: fatal: unable to parse config file '/home/modocache/GitHub/apple/swift/test/lit.cfg', traceback: Traceback (most recent call last):
#   File "/home/modocache/GitHub/apple/llvm/utils/lit/lit/TestingConfig.py", line 101, in load_from_path
#     exec(compile(data, path, 'exec'), cfg_globals, None)
#   File "/home/modocache/GitHub/apple/swift/test/lit.cfg", line 743, in <module>
#     android_ndk_path = config.android_ndk_path
# AttributeError: TestingConfig instance has no attribute 'android_ndk_path'
```

Alternatively:

```py
# swift/test/lit.cfg
android_ndk_path = getattr(config, 'android_ndk_path')

# lit.py: TestingConfig.py:114: fatal: unable to parse config file '/home/modocache/GitHub/apple/swift/test/lit.cfg', traceback: Traceback (most recent call last):
#   File "/home/modocache/GitHub/apple/llvm/utils/lit/lit/TestingConfig.py", line 101, in load_from_path
#     exec(compile(data, path, 'exec'), cfg_globals, None)
#   File "/home/modocache/GitHub/apple/swift/test/lit.cfg", line 743, in <module>
#     android_ndk_path = getattr(config, 'android_ndk_path')
# AttributeError: TestingConfig instance has no attribute 'android_ndk_path'
```

Bizarrely, running `lit` directly works:

```
$ llvm/utils/lit/lit.py -v build/Ninja-ReleaseAssert/swift-linux-x86_64/test-android-armv7
```

I'm still trying to figure this out, I'm sure it's something silly I'm overlooking 😅